vcenter_operator/vcenter_sso.py

```diff
@@ -0,0 +1,205 @@
+                        retries,
+                    )
+                    raise SSOSkippedError()
+
+        try:
+            auth = sso.SsoAuthenticator(f"https://{host}/sts/STSService/{self.domain}")
+
+            saml_token = auth.get_bearer_saml_assertion(self.ad_ttu_username, self.ad_ttu_password, delegatable=True)
+
+            stub = pyVmomi.SoapStubAdapter(
+                host=host,
+                port=443,
+                version=SSO_VERSION,
+                path=f"/sso-adminserver/sdk/{self.domain}",
+                samlToken=saml_token,
+                poolSize=0,
+            )
+
+            session_manager = pyVmomi.sso.SsoSessionManager("ssoSessionManager", stub=stub)
+            session_manager.Login()
+
+            # we do not need to send the token anymore - makes for smaller requests
+            stub.samlToken = None
+
+            api = pyVmomi.sso.SsoAdminServiceInstance("SsoAdminServiceInstance", stub=stub).SsoAdminServiceInstance()
+            self.sso_admin_instances[host] = {"api": api, "retry": 0, "last_retry": time.time()}
+        except Exception as e:
+            LOG.error("Error connecting to vCenter SSO instance: %s", e)
+            if host in self.sso_admin_instances:
+                self.sso_admin_instances[host]["retry"] += 1
+                self.sso_admin_instances[host]["last_retry"] = time.time()
+            else:
+                self.sso_admin_instances[host] = {"api": None, "retry": 0, "last_retry": time.time()}
+            LOG.debug(
+                "Failed to connect to SSO instance %s. Incrementing retry count to %s.",
+                host,
+                self.sso_admin_instances[host]["retry"],
+            )
+            raise SSOSkippedError()
+
+    def list_service_users(self, host, search_string, limit=10000):
+        """List service-users in the vCenter via SSO instance"""
+        api = self._get_api_instance(host)
+
+        try:
+            principal_discovery_service = api.principalDiscoveryService
+            criteria = pyVmomi.sso.AdminPrincipalDiscoveryServiceSearchCriteria(
+                searchString=search_string, domain=self.domain
+            )
+            users = principal_discovery_service.FindUsers(criteria=criteria, limit=limit)
+            user_names = [user.id.name for user in users]
+        except Exception as e:
+            LOG.error("Error listing service-users: %s", e)
+            del self.sso_admin_instances[host]
+            raise SSOSkippedError()
+
+        return user_names
+    
+    def check_users_in_group(self, host, search_string, limit=10000):
+        """Checks if service-user is in the Administrators group in the vCenter via SSO instance"""
+        api = self._get_api_instance(host)
+
+        try:
+            principal_discovery_service = api.principalDiscoveryService
+            criteria = pyVmomi.sso.AdminPrincipalDiscoveryServiceSearchCriteria(
+                searchString="", domain=self.domain
```

Comment:
Is it possible to use `searchString="Administrators"` and thus filter the groups already on vCenter side?